R/repositories.R

```diff
@@ -185,6 +211,16 @@
 #' use of CRAN packages not consistent with _Bioconductor_ best
 #' practices.
 #'
+#' To install binary packages on Linux, a default binary package location url
+#' can be set with the `BIOCONDUCTOR_CONTAINER_BINARY_URL` environment variable.
+#' This environment variable can point to one of a couple of locations:
+#' 1) Google - https://storage.googleapis.com/bioconductor_docker
+#' 2) Azure - https://bioconductordocker.blob.core.windows.net
```

Comment:
Before merging, I think it would be good to get a final decision on an abstraction, I think I'd vote for `https://bioconductor.org/packages/<version>/container-binaries` with `.htaccess` redirects in bioconductor.org, but this is a decision for Vince... (internal slack link https://devteam-bioc.slack.com/archives/C1TULTW5N/p1629298678009900). Maybe this actually simplifies things -- just hard-code this as the default, and rely on the container to provide the `BIOCONDUCTOR_DOCKER_VERSION`.

Reply:
I guess we want a way for the user to 'opt out' of using binary installations, and I wonder whether a slightly different approach is to change the default `site_repository=` argument to `BiocManager::repositories()` to (a slightly tweaked, to return `character(0)` if container binaries are not appropriate) `.repositories_container_binaries()`, or maybe an option `container_binaries = ...` or ...?

Reply:
Hi Martin, @mtmorgan 
I've updated the code, tests, and docs. I wasn't sure how to implement the package global option. For now, I've added a quick override.